src/components/Card/CartProductCard/CartProductCard.jsx

```diff
@@ -1,49 +1,98 @@
 import React from "react";
 import { BiHeartFill, BiTrashFill } from "../../../assets/icons";
+import { useCartWishlist } from "../../../context";
+import { isInWishlist } from "../../../utils/cart-and-wishlist-functions";
 import "./cart-product-card.css";
 
 export function CartProductCard({
-  badge,
-  img,
-  isBtnSaveToWishlist = false,
-  originalPrice,
-  name,
-  discountedPrice,
-  discountPercent,
-  icon,
-  ...rest
+  product,
+  product: {
```

Comment:
appreciation - i didn't know this way of having the whole object and it's de-structured values too.